src/gradualizer_lib.erl

```diff
@@ -83,58 +84,86 @@ reverse_graph(G) ->
     from_edges(maps:keys(G), [ {J, I} || {I, Js} <- maps:to_list(G), J <- Js ]).
 
 
+get_user_type_definition(Types, Anno, Name, Args) ->
+    %% Let's check if the type is a known remote type.
+    case typelib:get_module_from_annotation(Anno) of
+        {ok, Module} ->
+            case gradualizer_db:get_opaque_type(Module, Name, Args) of
```

Comment:
I just saw you're using `get_opaque_type` here. I don't think want to do exhaustiveness-checking on opaque remote types, since they are opaque...

Reply:
Indeed, this was just an easy to way to avoid the `opaque` return value from `gradualizer_db:get_type`, which can be handled easily in `refinable` (`opaque` is not refinable), but [will require an error to be thrown if it ever shows up when returned in `pick_value`](https://github.com/erszcz/Gradualizer/blob/exhaustive-user-type/src/gradualizer_lib.erl#L181-L182). I wanted to avoid introducing a new kind of error as I don't see anything relevant to `opaque` in `gradualizer_fmt`.

Reply:
Good point. If we want to pick_value of type `{tag, opaque_type()}`, we'd need a value of the opaque type. But OTOH opaque values shouldn't be shown.

Since pick_value is only used for producing error messages, perhaps we can put an underscore or three dots in there instead? `{tag, _}` or `{tag, ...}`, whatever is best. We could also change the exhaustiveness error messages slightly so that it doesn't explicitly say that it's a value of the type. Just some idea. WDYT?